pkg/webhook/pod/mutating/persistent_pod_state.go

```diff
@@ -41,8 +42,13 @@ func (h *PodCreateHandler) persistentPodStateMutatingPod(ctx context.Context, re
 		req.AdmissionRequest.Resource.Resource != "pods" {
 		return true, nil
 	}
+
+	whiteList, err := configuration.GetPPSWatchWatchCustomWorkloadWhiteList(h.Client)
+	if err != nil {
+		return false, err
+	}
 	ref := metav1.GetControllerOf(pod)
-	if ref == nil || ref.Kind != "StatefulSet" {
+	if ref == nil || (ref.Kind != "StatefulSet" && !whiteList.ValidateAPIVersionAndKind(ref.APIVersion, ref.Kind)) {
```

Comment:
If Custom Kind StatefulSet but not in whitelist, i think should skip the webhook.